src/tests/ftest/util/command_utils_base.py

```diff
@@ -478,6 +479,10 @@ def __init__(self, namespace, command, path=""):
         super().__init__(namespace)
         self._command = command
         self._path = path
+        self._python = None
+        if self.command.endswith('.py'):
+            # Run python scripts with the python command
+            self._python = os.path.basename(sys.executable)
```

Comment:
From the logs this will be `python3`, I suspect `f"python{sys.version_info.major}.{sys.version_info.minor}"` would be safer, with the correct imports of course.

Reply:
I'm not sure we can rely on `f"python{sys.version_info.major}.{sys.version_info.minor}"` existing on remote nodes either. E.g. if `python3` is a symlink to a python installed in a non-standard directory, which I have seen before.
IMO the most proper thing is to setup the virtual env on all nodes or put it in a shared directory. Even if `pythonX.Y` exists on the remote node, it possibly doesn't have the necessary pip packages anyway.
But that's more work so I think we should just keep it simple for now and cross that bridge when we need to

Reply:
Out of scope for this PR but:

> IMO the most proper thing is to setup the virtual env on all nodes or put it in a shared directory. Even if pythonX.Y exists on the remote node, it possibly doesn't have the necessary pip packages anyway.

When there's a need for these helper/remote python scripts to access that then yes they should.  At that point the front-end node should pip install what it needs, then save the versions to a file and have the remote nodes all install from that file rather than from the requirments-ftest.txt file directly.  On that node CI should probably save that data as an artefact anyway.

A bigger problem for now is that the `util` directory has an identity crisis, it's mostly used by ftest but the user (I think still) needs to set `PYTHONPATH` in order to access it, and that itself messes up pylint testing which we have a hack in place for, then there's are a load of apparent circular dependencies that we have to ignore as well.  Remote python scripts which are run by tests on test nodes should not reside in the same directory in the filesystem tree, that just obfuscates and confuses things further IMO.

Reply:
> When there's a need for these helper/remote python scripts to access that then yes they should ...

I agree and think we could leave it at that until we need better handling, since it's only a couple oddball tests that need it anyway. I'm okay with whatever works for now.

Reply:
Yeah, we don't really install or setup ftest as a module. It's just a collection of python files that "works"